### PR TITLE
refactor(files): Fix some ESLint and Typescript issues in FilesList

### PR DESCRIPTION
**Pull request by susnux**
## Summary

* Do not use names with `_` prefix on Vue instances (reserved by Vue)
* Make translation function calls more consistent by directly use the `t` function instead of calling it on the instance
* Fix typing of `this.promise`

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/views/FilesList.vue [hunk 1]
```diff
@@ -481,7 +499,7 @@ export default defineComponent({
 			}
 
 			// If we have a cancellable promise ongoing, cancel it
-			if (typeof this.promise?.cancel === 'function') {
+			if (this.promise && 'cancel' in this.promise && typeof this.promise.cancel === 'function') {
```
Comment (ShGKme) [inline apps/files/src/views/FilesList.vue:502]: Why?

File: apps/files/src/views/FilesList.vue [hunk 2]
```diff
@@ -515,8 +533,10 @@ export default defineComponent({
 
 				// Update paths store
 				const folders = contents.filter(node => node.type === 'folder')
-				folders.forEach(node => {
-					this.pathsStore.addPath({ service: currentView.id, fileid: node.fileid, path: join(dir, node.basename) })
+				folders.forEach((node) => {
+					if (node.fileid !== undefined) {
+						this.pathsStore.addPath({ service: currentView.id, fileid: node.fileid, path: join(dir, node.basename) })
+					}
 				})
```
Comment (ShGKme) [inline apps/files/src/views/FilesList.vue:540]: Just to understand it better, in which case `fileid` is `undefined` here?

File: apps/files/src/views/FilesList.vue (hunk 1 above)
Comment (susnux) [reply apps/files/src/views/FilesList.vue:502]: `typeof this.promise?.cancel === 'function'` does not infer that `this.promise` is of type `CancelablePromise`.
You need to use `'cancel' in this.promise`, but this is invalid if `this.promise` is `undefined`.

Screenshot_20240522_161035

Comment (susnux) [reply apps/files/src/views/FilesList.vue]: So I refactored this a bit (now) to make things more clear.

**Review by skjnldsv — APPROVED** [review]

**Review by ShGKme — APPROVED** [review]

**Pull request merged**